src/validation.cpp

```diff
@@ -2730,9 +2724,7 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
                     if (!cache.HasICXSubmitEXTHTLCOpen(dfchtlc->offerTx))
                     {
                         CTokenAmount makerDeposit{DCT_ID{0}, offer->takerFee};
-                        auto res = cache.AddBalance(order->ownerAddress, makerDeposit);
-                        if (!res)
-                            LogPrintf("Can't refund makerDeposit back to owner: %s\n", res.msg);
+                        cache.AddBalance(order->ownerAddress, makerDeposit);
```

Comment:
Why skip the result check?

Reply:
Since we can subtract we can add, it's safe. If add fails that's mean corrupt storage.

Reply:
I think we shouldn't do close order, offer, dfchtlc, exthtlc if Sub/AddBalance failed!

Reply:
We should since if we not do it, order / offer and so on we have them in inconsistent state, they are not closed.